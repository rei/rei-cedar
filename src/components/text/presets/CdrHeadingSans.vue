<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import { baseTextProps } from '../../../types/interfaces';

defineOptions({
  name: 'CdrHeadingSans',
});

interface headingSansTextProps extends baseTextProps {
   /** 
     * Sets the type scale
     * @type scaleValue
     * @values 1,2,3
   */
  scale?: '1'|'2'|'3',

}

const props = withDefaults(defineProps<headingSansTextProps>(), {
  tag: 'h2',
  scale: '3',
});

const typeProperties = computed(() => {
  return {
    '--cdr-heading-sans-font-size': `var(--cdr-type-scale-${props.scale})`,
    '--cdr-heading-line-height': `var(--cdr-line-height-ratio-heading-${props.scale})`
  }
});

const baseClass = 'cdr-heading-sans';
const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
    :style="typeProperties"
  >
    <slot />
  </component>
</template>

<style module src="./styles/CdrHeadingSans.module.scss" lang="scss">
</style>  