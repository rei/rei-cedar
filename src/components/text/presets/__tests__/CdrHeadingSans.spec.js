import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrHeadingSans from '../CdrHeadingSans.vue';


describe('CdrHeadingSans', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingSans, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingSans, {
        props: { scale: '1' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
