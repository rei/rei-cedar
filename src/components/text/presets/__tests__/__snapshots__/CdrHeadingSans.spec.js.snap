// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrHeadingSans > default configuration > renders correctly 1`] = `
<h2
  class="cdr-heading-sans"
  style="--cdr-heading-sans-font-size: var(--cdr-type-scale-3); --cdr-heading-line-height: var(--cdr-line-height-ratio-heading-3);"
>
  
  paragraph
  
</h2>
`;

exports[`CdrHeadingSans > passing a scale > renders correctly 1`] = `
<h2
  class="cdr-heading-sans"
  style="--cdr-heading-sans-font-size: var(--cdr-type-scale-1); --cdr-heading-line-height: var(--cdr-line-height-ratio-heading-1);"
>
  
  paragraph
  
</h2>
`;
